AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1515,40 +1515,36 @@ - (void)_removeSubnode:(ASDisplayNode *)subnode
   [subnode __setSupernode:nil];
 }
 
+// NOTE: You must not called this method while holding the receiver's property lock. This may cause deadlocks.
 - (void)removeFromSupernode
 {
   ASDisplayNodeAssertThreadAffinity(self);
-  BOOL shouldRemoveFromSuperviewOrSuperlayer = NO;
+  _propertyLock.lock();
+    __weak ASDisplayNode *supernode = _supernode;
+    __weak UIView *view = _view;
+    __weak CALayer *layer = _layer;
+    BOOL layerBacked = _flags.layerBacked;
+  _propertyLock.unlock();
```

Comment:
Unfortunately we can't use a scoped locker because we want those variables (supernode, view, layer, etc) to exist in the larger scope. So I could declare them before the scope and then assign them new values but it'd be kind of messy. 

Since I don't call any methods in this scope, no exception can be thrown so I think we're good. Agreed?